universe/user/user.go

```diff
@@ -141,6 +145,40 @@ func (u *User) GetUserType() universe.UserType {
 	return u.userType
 }
 
+func (u *User) SetOfflineTimer() (bool, error) {
+	u.mu.RLock()
+	defer u.mu.RUnlock()
+
+	u.offlineTimer.Set(u.id, time.Minute*20, u.DeleteTemporaryUser)
+	u.log.Infof("Timer set: %s", u.GetID())
+	return true, nil
+}
+
+func (u *User) IsTemporaryUser() (bool, error) {
+	guestUserTypeID, err := common.GetGuestUserTypeID()
+	if err != nil {
+		return false, errors.WithMessage(err, "failed to get guestUserTypeID")
+	}
+	if u.GetUserType().GetID() != guestUserTypeID {
+		return false, nil
+	}
+
+	return true, nil
+}
```

Comment:
if this method or `common.GetGuestUserTypeID` has some fallback logic and couldn't return `err` it would simplify logic in several places